### PR TITLE
App page secrets error

### PR DESCRIPTION
**Pull request by AlinaGoaga**
Closes [#3843](https://app.zenhub.com/workspaces/team-pesto-6127680175876a0014fc5657/issues/gh/weaveworks/weave-gitops/3843)

**What changed?**
For helm releases, do not try and read secret if `spec.kubeConfig` has been set.

**Why was this change made?**
On the Applications Page, users would get several errors related to the secret not being present.
`get helmrelease inventory: secrets "sh.helm.release.v1.grafana-agent-logs.v7" not found`. 
This is not relevant for cases in which `kubeConfig` is used.

### CODE REVIEW

File: core/server/inventory.go [hunk 1]
```diff
@@ -195,8 +195,13 @@ func getHelmReleaseObjects(ctx context.Context, k8sClient client.Client, helmRel
 		Namespace: storageNamespace,
 	}
 
-	if err := k8sClient.Get(ctx, key, storageSecret); err != nil {
-		return nil, err
+	if helmRelease.Spec.KubeConfig != nil {
+		// helmrelease secret is on another cluster so we cannot inspect it to figure out the inventory and version and other things
+		return nil, nil
+	} else {
```
Comment (foot) [inline core/server/inventory.go:201]: Don't need the `else` as we're immediately returning in the above condition

File: core/server/inventory.go [hunk 2]
```diff
@@ -195,8 +195,13 @@ func getHelmReleaseObjects(ctx context.Context, k8sClient client.Client, helmRel
 		Namespace: storageNamespace,
 	}
 
-	if err := k8sClient.Get(ctx, key, storageSecret); err != nil {
-		return nil, err
+	if helmRelease.Spec.KubeConfig != nil {
```
Comment (foot) [inline core/server/inventory.go:198]: Maybe add a test for this function w/ a spec.kubeconfig

**Review by foot — COMMENTED** [review]
Nice clean up too!

File: core/server/inventory_test.go [hunk 3]
```diff
@@ -270,3 +271,63 @@ func TestGetInventoryHelmRelease(t *testing.T) {
...
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "config-map",
+			Namespace: ns.Name,
+		},
+		Data: map[string]string{
+			"key": "value",
+		},
+	}
```
Comment (foot) [inline core/server/inventory_test.go:317]: Maybe we can remove the configmap here?

**Comment by foot** [comment]
Lets try adding a little message to the UI too, some little admonition like..

> **Note**
> spec.Kubeconfig is set on this HelmRelease. Details about reconciled objects are not available.

**Comment by AlinaGoaga** [comment]
Hi @mmoulian, we're got an info alert in this PR. I can't seem to be able to find something similar in the designs. Would you mind having a look at the styles and let me know which colors you'd prefer / add it to Figma? Below is how it looks atm.

Thank you

Screenshot 2023-09-20 at 15 38 59

Screenshot 2023-09-20 at 15 38 51

**Comment by mmoulian** [comment]
Hi @AlinaGoaga! Searching through our archive files I found this solution. We did it for alerts and errors, but I'm not sure if the alert in PR is the same type or needs a distinction.

differents states

Link Figma: https://www.figma.com/file/IVHnM9iyeFWpd11evtY8ux/Weave-GitOps?type=design&node-id=19837%3A54217&mode=design&t=TuqsDZiQ4rVgsarD-1

This solution is applied in our Demo on a page like this:
Captura de pantalla 2023-09-20 a las 16 10 38

Let me know if it help you!

**Comment by AlinaGoaga** [comment]
Thanks @mmoulian, yup I know these ones, we need one in blue for the info box please (light and dark)

**Review by foot — APPROVED** [review]

**Comment by mmoulian** [comment]
@AlinaGoaga I made the design here. ..It is very similar to your proposal. You can see it in dark or light mode (see movie).
Things to keep in mind:
-The corners are rounded 8 px
-Texts 14 px / line height 16 px
-I added these colors to our Variable Design System (See image)

https://github.com/weaveworks/weave-gitops/assets/95616291/36d928a5-9d94-4b3e-8c4d-9c57ac8056c3
https://www.figma.com/file/IVHnM9iyeFWpd11evtY8ux/Weave-GitOps?type=design&node-id=17897%3A199195&mode=design&t=fxC0g6SpQsA2Cyie-1


https://github.com/weaveworks/weave-gitops/assets/95616291/5ae913bf-5f5a-43be-a697-caced482176d



Captura de pantalla 2023-09-21 a las 15 37 40

**Comment by AlinaGoaga** [comment]
Thanks @mmoulian, will open a small PR separately to adjust the style :)

File: ui/components/AutomationDetail.tsx [hunk 4]
```diff
@@ -87,10 +91,15 @@ function AutomationDetail({
       component: () => {
         return (
           <RequestStateHandler loading={isLoading} error={error}>
-            <ReconciledObjectsTable
-              className={className}
-              objects={data?.objects}
-            />
+            {automation.type === "HelmRelease" &&
+            (automation as HelmRelease).kubeConfig === "" ? (
```
Comment (rudolfdobias) [inline ui/components/AutomationDetail.tsx:95]: Seems that this change caused that only HelmRelease type can have detail now as reported in [this bug](https://github.com/weaveworks/weave-gitops/issues/4057).

**Pull request merged**